File: 백준/Silver/1026. 보물/보물.py
N = int(input())
A = list(map(int, input().split()))
B = list(map(int, input().split()))

A.sort()

B_indices = sorted(range(N), key=lambda i: -B[i])

S = 0
for i in range(N):
    S += A[i] * B[B_indices[i]]

print(S)

File: 백준/Silver/1026. 보물/보물.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[] A = new int[N];
        Integer[] B = new Integer[N];

        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }

        for (int i = 0; i < N; i++) {
            B[i] = sc.nextInt();
        }

        Arrays.sort(A); 

        Integer[] BIndices = new Integer[N];
        for (int i = 0; i < N; i++) {
            BIndices[i] = i;
        }

        Arrays.sort(BIndices, (i, j) -> B[j] - B[i]); 

        int S = 0;
        for (int i = 0; i < N; i++) {
            S += A[i] * B[BIndices[i]];
        }

        System.out.println(S);
    }
}